react/components/Modal/index.tsx

```diff
@@ -0,0 +1,281 @@
+
+export const ModalOverlay: FC<OverlayProps> = ({
+  isOpen,
+  onClose,
+  centered,
+  container,
+  closeOnOverlayClick,
+  onCloseTransitionFinish,
+  children,
+}) => {
+  const [showPortal, setShowPortal] = useState(isOpen)
+
+  useLayoutEffect(() => {
+    if (isOpen) setShowPortal(isOpen)
+  }, [isOpen])
+
+  const handleClick = () => {
+    if (!closeOnOverlayClick) return
+    onClose()
+  }
+
+  const handleAnimationEnd = () => {
+    if (isOpen) return
+    onCloseTransitionFinish?.()
+    setShowPortal(false)
+  }
+
+  return showPortal
+    ? createPortal(
+        /** This rule can be disabled because we are not using the onClick property to click
+         * the element itself but to capture outside clicks */
+        // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+        <div
+          className={classNames(
+            'flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill',
+            isOpen ? styles.openAnimation : styles.closeAnimation,
+            {
+              'items-start': !centered,
+              'items-center': centered,
+            }
+          )}
+          tabIndex={-1}
+          onClick={handleClick}
+          onKeyDown={() => {}}
+          onAnimationEnd={handleAnimationEnd}
+        >
+          <FocusLock className={`${styles.contents}`}>{children}</FocusLock>
+        </div>,
+        container
+      )
+    : null
+}
+
+const ModalContent = forwardRef<HTMLDivElement, ContentProps>(
+  function ModalContent(
+    {
+      onClose,
+      showCloseIcon,
+      title,
+      showTopBar,
+      bottomBar,
+      showBottomBarBorder,
+      responsiveFullScreen,
+      size,
+      closeOnEsc,
+      children,
+    },
+    forwardedRef
+  ) {
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+      if (!closeOnEsc || event.key !== 'Escape') {
+        return
+      }
+      event.stopPropagation()
+      onClose()
+    }
+    return (
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+      <div
+        className={classNames(
+          'flex flex-column relative bg-white shadow-5 center mv9',
+          {
+            'vw-40-ns': size === 'small',
+            'vw-50-ns': size === 'medium',
+            'vw-60-ns': size === 'large',
+            'h-100 h-auto-ns vw-100': responsiveFullScreen,
+            'vw-50': !responsiveFullScreen,
+          }
+        )}
+        onClick={e => e.stopPropagation()}
+        role="dialog"
+        onKeyDown={handleKeyDown}
+        ref={forwardedRef}
+      >
+        <TopBar
+          showCloseIcon={showCloseIcon}
+          onClose={onClose}
+          showTopBar={showTopBar}
+          responsiveFullScreen={responsiveFullScreen}
+        >
+          {title}
+        </TopBar>
+        <div
+          className={`ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 ${styles.scrollBar}`}
+          style={{ maxHeight: '60vh' }}
```

Comment:
You can create a token for this in your CSS since you have one.

Reply:
Didn't notice this one, thanks!